lib/tasks/trends.rake

```diff
@@ -29,4 +29,57 @@ namespace :trends do
     end
   end
 
+  desc 'Reduces amount of points for each trend group for the last month'
+  task :reduce => :environment do
+    #get the records that sould remain in the DB - should'n be too much of them
+    change_points = TrendCounter.unscoped.select("trend_id, trend_counters.count, min(created_at) as created_at").group("trend_id, count")
```

Comment:
this actually misses a few points, as it will care for all counters with the same value regardless of trend,e.g.

```___|---|___``` kind of graph will count both values from left and right incorrectly.

IMHO you should use something like (pseudo code)
for each trend, find the starting point, then search for the first trend that has a different counter.
once you find that, mark the start and end dates, create a new counter entry.
loop until you there are no more trend counters.